org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/labeling/XbaseImageAdornments.java

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * Copyright (c) 2013, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.xbase.ui.labeling;
+
+import static org.eclipse.jdt.ui.JavaElementImageDescriptor.*;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.common.types.JvmConstructor;
+import org.eclipse.xtext.common.types.JvmDeclaredType;
+import org.eclipse.xtext.common.types.JvmExecutable;
+import org.eclipse.xtext.common.types.JvmField;
+import org.eclipse.xtext.common.types.JvmIdentifiableElement;
+import org.eclipse.xtext.common.types.JvmOperation;
+import org.eclipse.xtext.common.types.util.DeprecationUtil;
+import org.eclipse.xtext.xbase.typesystem.override.OverrideHelper;
+
+import com.google.inject.Inject;
+
+/**
+ * Used for calculating the Image adornments.
+ */
+public class XbaseImageAdornments {
+
+	@Inject
+	private OverrideHelper overrideHelper;
+
+	/**
+	 * @return an {@code Integer} value representing the adornment to be rendered for a given {@code identifiableElement}
+	 */
+	public int get(JvmIdentifiableElement identifiableElement) {
+		if (identifiableElement instanceof JvmConstructor) {
+			return _get((JvmConstructor) identifiableElement);
+		} else if (identifiableElement instanceof JvmOperation) {
+			return _get((JvmOperation) identifiableElement);
+		} else if (identifiableElement instanceof JvmField) {
+			return _get((JvmField) identifiableElement);
+		} else if (identifiableElement instanceof JvmDeclaredType) {
+			return _get((JvmDeclaredType) identifiableElement);
+		} else if (identifiableElement != null) {
+			return _get(identifiableElement);
+		} else {
+			return _get((Void) null);
+		}
+	}
+
+	protected int _get(JvmDeclaredType jvmDeclaredType) {
+		int abstract0 = or(0, jvmDeclaredType.isAbstract(), ABSTRACT);
```

Comment:
Feels weird to have `or` in Java as a method, but I understand the reasoning behind it.